src/api_client/api_resource.js

```diff
@@ -37,19 +37,23 @@ export default class APIResource {
 	}
 
 	request(req, retry, retryCount = 1) {
-		const request = popsicle
+		const request = axios
 			.request(req)
-			.use(popsicle.plugins.parse(['json', 'urlencoded']))
-			.then(res => {
-				if (res.status >= 300) {
-					if (res.body && res.body.message) {
-						throw new Error(`Status ${res.status} : ${res.body.message}`);
+			.then(res => res.data)
+			.catch(error => {
+				if (error.response) {
+					if (error.response.data && error.response.data.message) {
+						throw new Error(
+							`Status ${error.response.status} : ${
+								error.response.data.message
+							}`,
+						);
 					}
 					throw new Error(
-						`Status ${res.status} : An unknown error has occurred.`,
+						`Status ${error.response.status} : An unknown error has occurred.`,
 					);
 				}
-				return res.body;
+				throw error;
 			});
 
 		if (retry) {
```

Comment:
Unrelated to this PR, but do we want to return on the line below?

Reply:
I think it's the same result.
if the `retry` is on, we want to attach the function to retry

Reply:
The issue I'm referring to is not whether the retry is carried out, but what the return value of `request` should be.